src/lints/enum_variant_deprecated.ron

```diff
@@ -0,0 +1,60 @@
+SemverQuery(
+    id: "enum_variant_deprecated",
+    human_readable_name: "enum variant marked with #[deprecated]", 
+    description: "An enum variant was marked with the #[deprecated] attribute.",
+    required_update: Minor,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        enum_name: name @output
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                        variant {
+                            variant_name: name @output @tag
+                            deprecated @filter(op: "=", value: ["$true"])  
+                            span_: span @optional {
+                                filename @output
+                                begin_line @output
+                            }
+                        }
```

Comment:
This block needs a `public_api_eligible` check, both here and on the baseline side.

Otherwise, this lint will flag variants that were `#[doc(hidden)]` in the baseline, which is incorrect since they weren't public API.

Relevant docs here:
https://github.com/obi1kenobi/trustfall-rustdoc-adapter/blob/rustdoc-v39/src/rustdoc_schema.graphql#L91-L110

Please make sure to also add tests for `#[doc(hidden)]` enums and variants, to make sure those are never flagged by this lint.